source/pervasive/map.rs

```diff
@@ -291,11 +286,11 @@ macro_rules! map_insert_rec {
 #[macro_export]
 macro_rules! map {
     [$($tail:tt)*] => {
```

Comment:
The inlining of the pattern doesn't quite work because we need `verus_proof_macro_exprs!` to reparse the token trees into rust expressions. Otherwise, literals like `2int` throw an error.

Unfortunately, `verus_proof_macro_exprs!` requires things to be a macro. There is `verus_proof_expr!` that doesn't have this requirement, but we need to perform the expr splitting and pass along to it, which doesn't quite work because patterns we write in `map` cannot yet know how verus wants the split to happen.

Thus, it appears that there still needs to be a single non-inlined point. The two different approaches for that non-inlined point are to either keep a separate `map_internal!` macro (with docs hidden), or to introduce a separate arm to the `map!` macro that is prefixed with something like an `__internal___` tag, which is equivalent but would show up in the docs.

Happy to switch it over to the `__internal__` tag version if that is what you were talking about (I don't have a preference on either).

Alternatively, it is possible I might have missed a different approach other than the approaches mentioned above (or maybe there is a better place to invoke `verus_proof_expr!`?); please do clarify if so.